core/src/processing/core/PShapeSVG.java

```diff
@@ -514,53 +514,86 @@ protected void parsePath() {
     char[] pathDataChars = pathData.toCharArray();
 
     StringBuilder pathBuffer = new StringBuilder();
-    boolean lastSeparate = false;
-    boolean isOnDecimal = false;
+
+    /*
+     * The state of the lexer:
+     * -1: just after the command (i.e. a single alphabet)
+     * 0: neutral state
+     * 1: on a digit sequence for integer representation
+     * 2: on a decimal
+     * 3: on a digit or a sign in exponent in scientific notation, e.g. 3.14e-2)
+     * 4: on a digit sequence in exponent
+     */
+    int lexState = 0;
```

Comment:
Understandable, could it be with a enum instead of an integer? I'm afraid we might lose this documentation at some stage